docs/docs/examples/workflow/multi_strategy_workflow.ipynb

```diff
@@ -0,0 +1,416 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #33.                ctx.data["judge"] = ReActAgent.from_tools(
</pre><p>Was there a specific reason for using a react agent as a judge, rather than an llm directly?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/run-llama/llama_index/pull/15445/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Reply:
My thinking was that since the agent is persistent it would remember the previous iterations of the query and not get stuck permanently in a loop of a bad query. In reality I had trouble coming up with any query "bad" enough to trigger the loop in the first place.

Reply:
Interesting point! That's fair

Reply:
If you want to remember past interactions, it could be more straightforward to use SimpleChatEngine (and chat engine with an llm, no tools, no indexes)

Reply:
Okay, done!